pylint/checkers/utils.py

```diff
@@ -1245,7 +1249,7 @@ def is_none(node: nodes.NodeNG) -> bool:
     )
 
 
-def node_type(node: nodes.NodeNG) -> Optional[type]:
+def node_type(node: nodes.NodeNG) -> Optional[Any]:
```

Comment:
Why return `Any`?

Reply:
Mypy was complaining about `type`. I tried to deduct what class `type` is but got stuck and added `Any`.

Reply:
`type` is the generic class type: https://docs.python.org/3/library/typing.html#typing.Type
It should work if you annotate `types = set()` with `Set[type]`.